.github/workflows/typecheck.yml

```
@@ -0,0 +1,21 @@
+name: TypeScript Checks
+
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, synchronize]
+    branches-ignore: [staging, production]
+
+jobs:
+  typecheck:
+    if: ${{ github.actor != 'OSBotify' || github.event_name == 'workflow_call' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
```

NAB but we've changed our policy for trusted developers such as GitHub and Ruby and you can use the version tag here:

```suggestion
      - uses: actions/checkout@v3
```